src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1591,7 +1591,7 @@ void TorrentHandle::handleTorrentFinishedAlert(const libtorrent::torrent_finishe
         m_moveFinishedTriggers.append(boost::bind(&Session::handleTorrentFinished, m_session, this));
     }
     else {
-        if (recheckTorrentsOnCompletion)
+        if (recheckTorrentsOnCompletion && !isUploading())
```

Comment:
a little bit weird to see `!isUploading()` here, can you check what is the value of `m_state` when the problem occurs?

Reply:
The value is of m_state is stalledUp that's why I chose isUploading().
Flow is like:
CheckFinishedAlert -> UpdateState (to stalledUp, which is correct) -> TorrentFinishAlert (here recheckOnCompletion triggers another check) -> Loop forever..
This matches my observations and the ones from the named issues as well. Happens when you want to seed a new torrent and have recheckOnCompletion on.

Reply:
It seems like your changes break "Recheck on completion" at all.

Reply:
yep doesn't work as expected. Still trying to understand how this issue occurs. To be continued..

Reply:
We need to make sure that the torrent was not in the "downloading" state before it was in the "checking" state...